Docs: Missing proto generation step in CONTRIBUTING.md causes errors for new contributors

### Plugin Type

VSCode Extension

### Cline Version

3.36.0

### What happened?

Problem
When following the setup instructions in CONTRIBUTING.md, running npm run watch from the terminal fails with multiple errors about missing proto files:

✘ [ERROR] Could not resolve "./shared/proto/host/window"
✘ [ERROR] Cannot read file: /path/to/src/shared/proto/cline/checkpoints
✘ [ERROR] Could not resolve "./shared/proto/host/window"✘ [ERROR] Cannot read file: /path/to/src/shared/proto/cline/checkpoints

Root Cause

The setup instructions don't mention running npm run protos to generate Protocol Buffer files before the first build. This step is required for the TypeScript compilation to work.

Current Behavior
Following the docs: npm run install:all → npm run watch = FAILS ❌
VS Code F5 works fine because tasks.json has proto generation as a dependency

Expected Behavior
Contributors should be able to run npm run watch successfully after following the setup instructions.

Proposed Solution
Update CONTRIBUTING.md to explicitly mention the proto generation step
Add an npm run dev script that runs protos + watch for easier terminal workflow
Consider adding these steps to make setup more discoverable

Impact
New contributors get errors immediately and may not know how to fix them without diving into the codebase.

### Steps to reproduce

1. Clone the repo
2. Following the docs: npm run install:all 
3. npm run watch = FAILS ❌



### Provider/Model

_No response_

### System Information

_No response_

--- a/.changeset/fix-contributing-proto-step.md
+++ b/.changeset/fix-contributing-proto-step.md
@@ -1,0 +1,6 @@
+---
+"claude-dev": patch
+---
+
+Docs: Add missing proto generation step in CONTRIBUTING.md and new `npm run dev` script for easier terminal workflow (fixes #7335)
+

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -46,7 +46,11 @@ We also welcome contributions to our [documentation](https://github.com/cline/cl
     ```bash
     npm run install:all
     ```
-4. Launch by pressing `F5` (or `Run`->`Start Debugging`) to open a new VSCode window with the extension loaded. (You may need to install the [esbuild problem matchers extension](https://marketplace.visualstudio.com/items?itemName=connor4312.esbuild-problem-matchers) if you run into issues building the project.)
+4. Generate Protocol Buffer files (required before first build):
+    ```bash
+    npm run protos
+    ```
+5. Launch by pressing `F5` (or `Run`->`Start Debugging`) to open a new VSCode window with the extension loaded. (You may need to install the [esbuild problem matchers extension](https://marketplace.visualstudio.com/items?itemName=connor4312.esbuild-problem-matchers) if you run into issues building the project.)
 
 
 
@@ -85,8 +89,10 @@ We also welcome contributions to our [documentation](https://github.com/cline/cl
 
 2. **Local Development**
     - Run `npm run install:all` to install dependencies
+    - Run `npm run protos` to generate Protocol Buffer files (required before first build)
     - Run `npm run test` to run tests locally
     - Run → Start Debugging or `>Debug: Select and Start Debugging` and wait for a new VS Code instance to open
+    - **Terminal Workflow**: Use `npm run dev` (generates protos + runs watch mode) or `npm run watch` (if protos already generated)
     - Before submitting PR, run `npm run format:fix` to format your code
 
 3. **Linux-specific Setup**

--- a/package.json
+++ b/package.json
@@ -304,6 +304,7 @@
 		"dev:cli:watch": "node scripts/dev-cli-watch.mjs",
 		"postcompile-standalone": "node scripts/package-standalone.mjs",
 		"postcompile-standalone-npm": "node scripts/package-standalone.mjs --target=npm",
+		"dev": "npm run protos && npm run watch",
 		"watch": "npm-run-all -p watch:*",
 		"watch:esbuild": "node esbuild.mjs --watch",
 		"watch:tsc": "tsc --noEmit --watch --project tsconfig.json",

